packages/livekit-rtc/src/room.ts

```diff
@@ -74,26 +75,44 @@ export class Room extends (EventEmitter as new () => TypedEmitter<RoomCallbacks>
     super();
   }
 
-  get name(): string {
-    return this.info.name;
+  get name(): string | undefined {
+    return this.info?.name;
   }
 
-  get metadata(): string {
-    return this.info.metadata;
+  get metadata(): string | undefined {
+    return this.info?.metadata;
   }
 
   get isConnected(): boolean {
     return this.ffiHandle != undefined && this.connectionState != ConnectionState.CONN_DISCONNECTED;
   }
 
   async getSid(): Promise<string> {
-    return this.info.sid; // TODO update this to handle async room updates once rust protocol has been updated
+    if (!this.isConnected) {
+      return '';
+    }
+    if (this.info && this.info.sid !== '') {
+      return this.info.sid;
+    }
+    return new Promise((resolve, reject) => {
```

Comment:
in order for this to work, we'll need to expose a FFI event for room info updates, so this PR is blocked by support for that.